src/audio/mixer.c

```diff
@@ -351,6 +371,11 @@ static int mixer_copy(struct comp_dev *dev)
 	/* write zeros if all sources are inactive */
 	if (num_mix_sources == 0) {
```

Comment:
according to the issue,  "num_mix_sources == 0" is not fit for the case. Can we change the condition to " if no available data"

Reply:
@RanderWang  @lyakh  I agree "num_mix_sources" is very confusing, but I didn't want to start addressing other issues in the same PR. I can take a look at this though. I wanted to keep the review easier, but given the comments, it seems num_mix_sources just add to the confusion, so maybe better to fix in the same PR.